pwiz_tools/Skyline/Model/SmallMoleculeTransitionListReader.cs

```diff
@@ -604,42 +604,45 @@ private int INDEX_DECLUSTERING_POTENTIAL
                 maxCharge, new int[0],
                 TransitionCalc.MassShiftType.none, out _, out _);
 
-            if (isPrecursor && adductInferred.IsEmpty)
+            if (adductInferred.IsEmpty)
             {
                 // See if this can be explained by the more common adduct types, possibly with water loss
                 var matches = new Dictionary<double, Adduct>();
-                foreach (var text in Adduct.DEFACTO_STANDARD_ADDUCTS)
+                foreach (var adductList in new []{Adduct.DEFACTO_STANDARD_ADDUCTS, Adduct.COMMON_CHARGEONLY_ADDUCTS})

```

Comment:
If you do:
`                foreach (var text in Adduct.DEFACTO_STANDARD_ADDUCTS.Concat(Adduct.COMMON_CHARGEONLY_ADDUCTS))
`
then you won't need two nested for loops.

Reply:
I considered that (I agree that less code is generally better code), but it seems like a lot of memory manipulation in code that can be called quite frequently - not that I actually tried to measure relative performance or anything.

Reply:
By the way, "Enumerable.Concat" does not actually have to create a single list containing all of the elements. Instead, it returns an Iterator which first enumerates all of the items in the first list and then the items in the second list.